db/model/room.js

```diff
@@ -75,22 +80,33 @@ module.exports = function room(seq, dataTypes) {
       },
     },
     hooks: {
-      afterCreate: (instance) => {
+
+      /**
+       * Ensures room gets default values from roomType, and publishes
+       * room updates to redis.
+       *
+       * @param {Instance} instance - The instance being created
+       * @returns {Promise} which resolves to the instance
+       */
+      beforeCreate: (instance) => {
         const RoomType = seq.models.RoomType;
         return RoomType.findById(instance.type)
         .then((roomType) => {
           instance.settings = roomType.settings;
+          instance.bots = roomType.bots;
           const changedKeys = Object.keys(instance._changed);
           const ignoreAttributes = ['isDeleted'];
-          return realTime.publishObject(instance.toJSON(), roomEventNames.add, changedKeys,
-              ignoreAttributes);
+          return realTime.publishObject(instance.toJSON(), roomEventNames.add,
```

Comment:
Seems weird to publish object BEFORE create. What if there's an error in the creation itself? Why are you doing it this way?